File: src/main/java/reports/ReportTester.java
package reports;

import java.util.ArrayList;
import java.util.List;

public class ReportTester {
    public static void main(String[] args) {
        Report financeReport = new FinanceReport();
        Report healthReport = new HealthReport();
        List<Report> reports = new ArrayList<>();
        reports.add(financeReport);
        reports.add(healthReport);
        for (Report n:reports){
            n.load();
            n.read();
        }
    }
}

File: src/main/kotlin/reports/ReportTester.kt
package reports.kotlin

class ReportTester() {
}
fun main(){
    val healthReport:Report = HealthReport()
    val financeReport:Report = FinanceReport()
    val reports:MutableList<Report> = mutableListOf()
    reports.add(healthReport)
    reports.add(financeReport)
    for (item in reports){
        item.load()
        item.read()
    }
}